unumpy/_multimethods.py

```diff
@@ -453,6 +453,48 @@ def asanyarray(a, dtype=None, order=None):
     return (a, mark_dtype(dtype))
 
 
+def _asfarray_default(a, dtype=float):
+    a = asarray(a, dtype=dtype)
+    if a.dtype.name not in ("float16", "float32", "float64"):
+        dtype = float
+
+    return asarray(a, dtype=dtype)
+
+
+@create_numpy(_dtype_argreplacer, default=_asfarray_default)
+@all_of_type(ndarray)
+def asfarray(a, dtype=float):
+    return (mark_dtype(dtype),)
+
+
+@create_numpy(
+    _dtype_argreplacer,
+    default=lambda a, dtype=None: asarray(a, dtype=dtype, order="F"),
+)
+@all_of_type(ndarray)
+def asfortranarray(a, dtype=None):
+    return (mark_dtype(dtype),)
+
+
+def _asarray_chkfinite_default(a, dtype=None, order=None):
+    if any(isnan(a) | isinf(a)):
```

Comment:
Slightly  more efficient this way:
```suggestion
    if not isfinite(a).all():
```

Also, the `asarray` call should really come first so `a` is only converted to an array once.

Reply:
I'd suggest `all(isfinite(a))`, even though it does incur a multimethod call, it's more robust.